### ISSUE TITLE
e var=? functionality is undocumented

### ISSUE THREAD

**Issue by ewtoombs** [radare/radare2#7127]
For example, `e search.in=?` lists all valid settings for `search.in`. `e?` does not document this pattern.

```
[0x81dffff8]> e search.in=?
raw
block
file
io.maps
io.maprange
io.section
io.sections
io.sections.write
io.sections.exec
dbg.stack
dbg.heap
dbg.map
dbg.maps
dbg.maps.exec
dbg.maps.write
anal.fcn
anal.bb
[0x81dffff8]> e asm.arch=?
_dAe  _8_16      6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
_dA_  _8         8051        PD      8051 Intel CPU
_dA_  _16_32     arc         GPL3    Argonaut RISC Core
a___  _16_32_64  arm.as      LGPL3   as ARM Assembler (use ARM_AS environment)
adAe  _16_32_64  arm         BSD     Capstone ARM disassembler
...
```

**Comment by radare** [comment]
It is documented in the description of the evalvar. See e??

**Comment by radare** [comment]
Feel free to send a pr adding this hint in the e? Help if you find it necessary

**Comment by ewtoombs** [comment]
I do plan on helping fix all the bugs I'm posting. I'm using this system to document all the bugs I'm finding as I use r2. When I'm done using r2 for this task, I intend to go back and fix everything, but right now, I just want to concentrate on getting this task done. I don't like switching back and forth between using software and fixing its bugs. If I don't document the bugs as I find them, though, I tend to forget to do so.

I looked at e?? and I didn't find any description of the general pattern of `e var=?`. Neither of the evalvar descriptions for the specific evalvars I posted (asm.arch, search.in) said anything like "to get the full list of options, type e var=?".
```
[0x00000000]> e?asm.arch
            asm.arch: Set the arch to be used by asm
[0x00000000]> e?search.in
           search.in: Specify search boundaries (raw, block, file, section, range)
```
On top of that, "section" and "range" are not present in `e search.in=?`.
```
[0x00000000]> e search.in=?
raw
block
file
io.maps
io.maprange
io.section
io.sections
io.sections.write
io.sections.exec
dbg.stack
dbg.heap
dbg.map
dbg.maps
dbg.maps.exec
dbg.maps.write
anal.fcn
anal.bb
[0x00000000]> 
```
